app/validators/answer_code_validator.py

```diff
@@ -0,0 +1,197 @@
+from app.validators.questionnaire_schema import find_duplicates
+from app.validators.validator import Validator
+
+
+class AnswerCodeValidator(Validator):
+    INCORRECT_DATA_VERSION_FOR_ANSWER_CODES = (
+        "Answer codes are only supported in data version 0.0.3"
+    )
+    DUPLICATE_ANSWER_CODE_FOUND = "Answer codes must be unique"
+    MISSING_ANSWER_CODE = "No answer codes found for answer_id set in the schema"
+    ANSWER_CODE_ANSWER_ID_NOT_FOUND_IN_SCHEMA = (
+        "No matching answer id found in the schema for the given answer code"
+    )
+    ANSWER_VALUE_SET_FOR_ANSWER_WITH_NO_OPTIONS = (
+        "Answer values can only be set for answers that support answer options"
+    )
+    ANSWER_CODE_MISSING_FOR_ANSWER_OPTIONS = (
+        "The number of answer codes does not match number of answer options"
+    )
+    MORE_THAN_ONE_ANSWER_CODE_SET_AT_PARENT_LEVEL = "Only one answer code should be set for an answer when not specifying answer codes for answer options"
+    INCORRECT_VALUE_FOR_ANSWER_CODE_WITH_ANSWER_OPTIONS = "Values specified in answer code and answer options do not match or they are of different lengths."
+    DYNAMIC_ANSWER_OPTION_MUST_HAVE_ANSWER_CODE_SET_AT_TOP_LEVEL = (
+        "Dynamic options must have an answer code set at the parent level"
```

Comment:
This isn't very clear on how to resolve. Maybe something like `Answers with dynamic options must have an answer code mapping without answer value`?. Var name needs to change accordingly. Parent level is confusing since the code isnt being defined at the answer level.